### ISSUE TITLE
[Header] GitHub link is missing an Accessible Name (accessibility bug)

### ISSUE THREAD

**Issue by ashleemboyer** [radix-ui/website#786]
### Current Behavior

1. Go to https://www.radix-ui.com
2. Open Chrome DevTools
3. Open the Elements panel
4. Locate the GitHub link element
5. Open the Accessibility panel
6. See that the Accessible Name is empty (`""`), like pictured:

![](https://github.com/radix-ui/website/assets/43934258/7b0a0240-ad5c-4f16-b32c-c08460ebbeaa)

### Expected behavior

The Accessible Name should match the tooltip that appears on hover/focus.

### Suggested solution

Set the `aria-label` on the element to match the text of the tooltip. This value should be set all the time, not just when the tooltip shows.

Location of the code, from what I can tell: https://github.com/radix-ui/website/blob/52578d3c5956b26c117ad8328ee40ecc6170b648/components/Header.tsx#L145-L153

Further improvement: update the tooltip text (and the link's aria-label) to be `"GitHub"`. Otherwise, speech recognition users (who can't open tooltips) have to guess until they say "View GitHub" in order to open the link.
